sflib.py

```diff
@@ -657,7 +668,7 @@ def configSerialize(self, opts, filterSystem=True):
             if type(opts[opt]) is list:
                 storeopts[opt] = ','.join(opts[opt])
 
-        if '__modules__' not in opts:
+        if opts['__modules__'] is None:
```

Comment:
This will generate an exception if `__modules__` doesn't exist. I haven't looked thoroughly but there may be cases where it doesn't exist, so would be better to leave as it was or use `.get()` instead.

Reply:
Good. It should error here. Better error handling would be great, but I would rather it die here due to a missing key rather than die several layers deeper resulting in an unintuitive stack trace like this (which is what happens with the original code or when using `get()`):

```
_____________________________________________________________ TestSpiderFootScanner.test_set_status_invalid_status_should_raise ______________________________________________________________

self = <test.unit.test_spiderfootscanner.TestSpiderFootScanner testMethod=test_set_status_invalid_status_should_raise>

    def test_set_status_invalid_status_should_raise(self):
        """
        Test def setStatus(self, status, started=None, ended=None)
        """
>       sfscan = SpiderFootScanner("", "", "IP_ADDRESS", list(), self.default_options)

test/unit/test_spiderfootscanner.py:86: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
sfscan.py:82: in __init__
    self.dbh.scanConfigSet(self.scanId, self.sf.configSerialize(deepcopy(self.config)))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <sflib.SpiderFoot object at 0x7f608b0df280>, opts = {'__blocknotif': False, '__database': 'spiderfoot.test.db', '__docroot': '', '__logging': True, ...}, filterSystem = True

    def configSerialize(self, opts, filterSystem=True):
        """Convert a Python dictionary to something storable in the database.
    
        Args:
            opts (dict): TBD
            filterSystem (bool): TBD
    
        Returns:
            dict: config options
        """
    
        if not isinstance(opts, dict):
            raise TypeError("opts is %s; expected dict()" % type(opts))
    
        storeopts = dict()
    
        if not opts:
            return storeopts
    
        for opt in list(opts.keys()):
            # Filter out system temporary variables like GUID and others
            if opt.startswith('__') and filterSystem:
                continue
    
            if type(opts[opt]) is int or type(opts[opt]) is str:
                storeopts[opt] = opts[opt]
    
            if type(opts[opt]) is bool:
                if opts[opt]:
                    storeopts[opt] = 1
                else:
                    storeopts[opt] = 0
            if type(opts[opt]) is list:
                storeopts[opt] = ','.join(opts[opt])
    
>       if not opts.get(['__modules__']):
E       TypeError: unhashable type: 'list'

sflib.py:671: TypeError
```

Reply:
```suggestion
        # todo: ensure the __modules__ key value is in the expected format
        if '__modules__' not in opts:
```